### ISSUE TITLE
Change route for admin url

### ISSUE THREAD

**Issue by jojouser** [microweber/microweber#277]
static admin url is a normal security issue.
Then how can i change the route for admin url?
For example,
  change from "domain.com/admin/" to "domain.com/adminnewurl/"
thanks.

**Comment by nickwuk** [comment]
Yes I need this too.

**Comment by ash-rain** [comment]
It sounds reasonable and we'll consider that for version 1.0 but being able to access the admin login is not exactly a security issue unless there are other factors involved. For the time being you can try tinkering https://github.com/microweber/microweber/blob/1.0/src/Microweber/routes.php
if you insist on changing the route but do so on your own risk.

**Comment by ash-rain** [comment]
By the way if you two are working on something using Microweber I'd love to hear more about it and help in any way I can. You can find my email in my profile.

**Comment by jojouser** [comment]
thanks. just waiting version 1.0.

**Comment by nickwuk** [comment]
One site built at fourpawshotel dot co dot uk and now waiting for version 1.0 before building more.

**Comment by ash-rain** [comment]
Great job! Hold on just a little longer. It's worth the wait

**Comment by shchukax** [comment]
I still don't see this feature in version 1.0.7 (it's March 2016)

**Comment by Staalkoper** [comment]
Probably because, as Ash pointed out, it isn't necessarily a security issue and thus it has been lost in the void of more pressing updates :)

**Comment by shchukax** [comment]
I disagree.  It is a security issue, because it's unnecessarily leaks information about the platform. Just seeing the login form, I know that the site is built on microweber and the deployed version of it.  Now I can get the source code of that version and specifically search for vulnerabilities that would let me get into the site.

I edited `routes.php` to change the url, however it's not enough.  There are 32 files inside `modules/admin` that have hardcoded url `/admin/<something>` - these are a combination of php and javascript files.  To make the complete change, I'll have to edit these all.  For now I left rout `/admin/{all}` and set it to be `'as' => 'new_admin_url'`, however this is bad security and I'm planning to change the url in all of those 32 files to my new URL.

I can see how this isn't a trivial change in all.  You can easily use a setting in PHP, however to then propagate the change into the javascript would require changing the javascript to be produced on the fly by php or another means to inject the value of the config parameter.

I'll keep watching this issue to see if it gets resolved.

**Comment by nickwuk** [comment]
I agree with shchukax. I always change the admin url whenever possible to make it one step more difficult for those trying to guess it's location, and then setup a honeypot IP trap in the admin directory. People/robots are looking for /admin or /wp-admin, etc all the time.

**Comment by peter-mw** [comment]
It is a needed option yes. 

I have to figure out how to make routes from a value stored in the laravel the config file and set the admin url value there

**Comment by shchukax** [comment]
Ok, I think I've done everything that's required now.  For those that are looking to do this, here's everything that I did.  Note that this is a _hack_, it's not the right way of solving this problem.  And you'll need to re-implement it if you upgrade your MW installation.  However this does the job for me.

1. Edit: `src/Microweber/routes.php`

```
21,22c21,22
-     Route::any('/admin', 'AdminController@index');
-     Route::any('admin', array('as' => 'admin', 'uses' => 'AdminController@index'));
---
+     Route::any('/newadmin', 'AdminController@index');
+     Route::any('newadmin', array('as' => 'newadmin', 'uses' => 'AdminController@index'));
24c24,25
-     Route::any('admin/{all}', array('as' => 'admin', 'uses' => 'AdminController@index'))->where('all', '.*');
---
+     Route::any('newadmin/{all}', array('as' => 'newadmin', 'uses' => 'AdminController@index'))->where('all', '.*');
>     Route::any('newadmin/{all}', array('as' => 'newadmin', 'uses' => 'AdminController@index'))->where('all', '.*');
```

2. Edit the default page template for your site to be whatever you want the 'not found' page to look like (this was a weird requirements), as I found out that out-of-the-box installation of microweber would simply send back the default page template whenever I request something that doesn't exist.

3. Edit `userfiles/modules/admin/index.php`:

2a3,11
  <?php if(is_admin() == false): ?>
+ <?php
+     $url = trim($_SERVER['REQUEST_URI'], '/');
+     $parts = preg_split('/[\/\?]/', $url);
+     if(count($parts) && $parts[0] != 'newadmin') {
+         header($_SERVER["SERVER_PROTOCOL"]." 404 Not Found");
+         $data = file_get_contents('http' . (empty($_SERVER['https']) ? '' : 's') . '://' . $_SERVER['HTTP_HOST'] . '/not-found');
+ 	echo $data;
+         exit;
+     }
> ?>

This last part is a trick to avoid editing any other files. Basically, what happens here is that 404 is returned if `/admin/something` is called directly without logging in first.  If you are logged in as an admin, then it'll let this through.  The other thing to note here is the `file_get_contents` of `not-found` page.  That page doesn't exist on my site, therefore, as described in point 2 above, the default page template will be returned. I'm simply reading the content of that default page and returning it along with the 404.

**Comment by peter-mw** [comment]
hi @shchukax this fix probably wont work for everything, because there are some urls that point to the /admin url and they will break

**Comment by shchukax** [comment]
@peter-mw I did a quick grep search for urls with `/admin` - the only ones I found were those that had `/admin/something` - I didn't see any that are to `/admin` without anything following it.  However if this is the case, then I can easily work around it by adding an extra line to `routes.php`

So far everything is working for me, however I haven't gone through all the admin functionality yet.

**Comment by shchukax** [comment]
Ah! Is this now available in 1.0.8?

**Comment by peter-mw** [comment]
yes 

in `config/microweber.php` you can set for example ` 'admin_url' => 'my_admin',` 


and then go on `http://localhost/my_admin`

**Comment by Ezyweb-uk** [comment]
Would it be worth adding this config option to the install screen?

**Comment by peter-mw** [comment]
@Ezyweb-uk good idea

**Comment by peter-mw** [comment]
added for version 1.1.12 that will be released soon
